main.sh

```diff
@@ -91,12 +92,27 @@ trap cleanup_before_exit EXIT
 
 # Translate usage string -> getopts arguments, and set $arg_<flag> defaults
 while read line; do
-  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-##')"
+  # fetch single character version of option sting
+  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-\(.\)#\1#')"
+
+  # fetch long version if present
+  long_opt="$(echo "${line}" |awk 'match($2,"^--.*",a){print substr(a[0],3)}')"
+  # map long name back to short name
+  varname="short_opt_${long_opt}"
+  eval "${varname}=\"${opt}\""
+  # map short name to long name ... I don't think this is needed
+  varname="long_opt_${opt:0:1}"
+  eval "${varname}=\"${long_opt}\""
+
+  # check if option takes an argument
+  varname="has_arg_${opt}"
   if ! echo "${line}" |egrep '\[.*\]' >/dev/null 2>&1; then
     init="0" # it's a flag. init with 0
+    eval "${varname}=0"
   else
     opt="${opt}:" # add : if opt has arg
     init=""  # it has an arg. init with ""
+    eval "${varname}=1"
```

Comment:
It seems that even before these changes, if a required option argument was missing, subsequent options/text would get parsed as the required argument. It might be sensible to check that required arguments aren't just being populated with additional options/arguments. See also #5